app/views/submissions/_code_viewer.html.erb

```diff
@@ -44,24 +44,28 @@
   <% else %>
     <div class="code-table">
         <% @data.each_with_index do |(code, annotation), index| %>
-          <div class="code-line" id="line-<%=index%>" data-line-id="<%=index%>">
-            <div class="line-sticky">
-              <div class="line-number"><%= index+1 %></div>
-              <div class="code-separator"></div>
+          <% if code.valid_encoding? %>
+            <div class="code-line" id="line-<%=index%>" data-line-id="<%=index%>">
+              <div class="line-sticky">
+                <div class="line-number"><%= index+1 %></div>
+                <div class="code-separator"></div>
+              </div>
+              <div class="add-comment">
+                <% if @cud.instructor? or @cud.course_assistant? then %><div class="add-button-container"><button class="add-button" id="add-btn-<%= index+1 %>"><i class="material-icons">add_comment</i></button></div><% end %>
+              </div>
+              <% begin %>
+              <div class="code"><pre><code class="<%= @displayFilename.split(".")[-1].downcase %>"><%= code %></code></pre></div>
```

Comment:
The issue seems to lie with the `<%= code %>` block.

Actually, what we could do is to replace it with `<%= code.encode("UTF-8", invalid: :replace, replace: "") %>`. That way, we can render what's left of the file.

What do you think?

Reply:
Seems like Japanese works fine, so I'm confused as to what characters specifically cause this UTF-8 error.
This is what it looks like when trying to render what's left of the file:
<img width="915" alt="Screen Shot 2022-10-06 at 21 21 40" src="https://user-images.githubusercontent.com/25730111/194447058-88c39fad-c68f-4949-9c42-d4eb713ebc48.png">
Rejecting it entirely:
<img width="913" alt="Screen Shot 2022-10-06 at 21 27 24" src="https://user-images.githubusercontent.com/25730111/194447061-f324b7b7-2569-43a9-8847-d0e42bf2a18e.png">

Which do you think looks better? Personally rejecting it looks better and I mean the data that is able to be rendered is not very useful.

Reply:
<img width="910" alt="Screen Shot 2022-10-06 at 21 29 21" src="https://user-images.githubusercontent.com/25730111/194447211-7ff146ab-8a52-4d48-9f23-a5d4513f24d6.png">
Japanese text rendering on nightly

Reply:
I think we can just reject the files then. However, we might want to update the message - the error is that the file contains non UTF-8 characters, rather than non-ASCII

Reply:
Gotcha, I'll update the display.